### PR TITLE
SNOW-1526508 Pick top level domain for Snowflake hosts

### PR DESCRIPTION
**Pull request by sfc-gh-pmotacki**
### Description
Please explain the changes you made here.

### Checklist
- [ ] Format code according to the existing code style (run `npm run lint:check -- CHANGED_FILES` and fix problems in changed code)
- [ ] Create tests which fail without the change (if possible)
- [ ] Make all tests (unit and integration) pass (`npm run test:unit` and `npm run test:integration`)
- [ ] Extend the README / documentation and ensure is properly displayed (if necessary)
- [ ] Provide JIRA issue id (if possible) or GitHub issue id in commit message

### CODE REVIEW

File: lib/agent/socket_util.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ const OcspResponseCache = require('./ocsp_response_cache');
 const Errors = require('../errors');
 const ErrorCodes = Errors.codes;
 
-const REGEX_SNOWFLAKE_ENDPOINT = /.snowflakecomputing.com$/;
+const REGEX_SNOWFLAKE_ENDPOINT = /.snowflakecomputing.(com|cn)?$/;
```
Comment (sfc-gh-dprzybysz) [inline lib/agent/socket_util.js:14]: it's not correct when there will be a different top level domain - may we just leave `/.snowflakecomputing./`?

File: lib/connection/connection.js [hunk 2]
```diff
@@ -20,7 +20,7 @@ const { init: initEasyLogging } = require('../logger/easy_logging_starter');
 const GlobalConfig = require('../global_config');
 const JsonCredentialManager = require('../authentication/secure_storage/json_credential_manager');
 
-const PRIVATELINK_URL_SUFFIX = '.privatelink.snowflakecomputing.com';
+const REGEX_PRIVATELINK_SUFFIX = /.privatelink.snowflakecomputing.(com|cn)?$/;
```
Comment (sfc-gh-dprzybysz) [inline lib/connection/connection.js:23]: it's not correct when there will be a different top level domain - may we just leave /.privatelink.snowflakecomputing./?

File: test/unit/connection/connection_config_test.js [hunk 3]
```diff
@@ -966,6 +966,77 @@ describe('ConnectionConfig: basic', function () {
...
+            account: 'account-123xyz',
+            region: 'cn-north-1'
+          },
+        options:
+          {
+            accessUrl: 'https://account-123xyz.cn-north-1.snowflakecomputing.cn',
+            username: 'username',
+            password: 'password',
+            account: 'account-123xyz',
+            region: 'cn-north-1'
```
Comment (sfc-gh-dprzybysz) [inline test/unit/connection/connection_config_test.js:984]: let's add test with uppercase

File: lib/connection/connection.js [hunk 4]
```diff
@@ -265,12 +266,15 @@ function Connection(context) {
    * @returns {Object} the connection object.
    */
   this.connectAsync = async function (callback) {
+    const connectingDomain = connectionConfig.accessUrl && connectionConfig.accessUrl.includes('snowflakecomputing.cn') ? 'CHINA' : 'GLOBAL';
```
Comment (sfc-gh-dprzybysz) [inline lib/connection/connection.js:269]: can we create a method to not duplicate the logic and log?

**Review by sfc-gh-dprzybysz — APPROVED** [review]

**Review by sfc-gh-dheyman-1 — APPROVED** [review]

File: lib/connection/connection.js (hunk 4 above)
Comment (sfc-gh-pmotacki) [reply lib/connection/connection.js:269]: Fixed

**Pull request merged**